connection.cpp

```diff
@@ -392,22 +470,22 @@ void UDPThread::transmitBuffer() {
   delete[] packetBuffer;
 }
 
-inline uint32_t UDPThread::getTimerValue() {
+inline uint32_t UDPThread::getTimerValue(int timerFd) {
   struct itimerspec ts;
-  timerfd_gettime(m_timerfd, &ts);
+  timerfd_gettime(timerFd, &ts);
   return ts.it_value.tv_sec*1000000 + ts.it_value.tv_nsec/1000;
 }
 
-inline void UDPThread::adjustTimer(uint32_t interval, uint32_t value) {
+inline void UDPThread::adjustTimer(int timerFd , uint32_t interval, uint32_t value) {
   struct itimerspec ts;
   /* Setting the value to 0 disables the timer */
   if (value == 0)
-    value = 1;
+    //value = 1;
```

Comment:
This needs to be fixed / checked. If a user passes a timeout of 0 in the .csv file to have a timeout of zero (no timeout) this will disable the timer.

So either fix it here:
https://github.com/mguentner/cannelloni/blob/3de30242201799d32f9b810973fc12f8eba42d9b/cannelloni.cpp#L158or here (I think this is better):
https://github.com/mguentner/cannelloni/blob/3de30242201799d32f9b810973fc12f8eba42d9b/connection.cpp#L291